doc/dev-guide/src/release-process.md

```diff
@@ -1,66 +1,217 @@
-# Making a release
+# Release Process
 
-Before making a release, ensure that `rustup-init.sh` is behaving correctly,
+This document describes the process for making a new release of Rustup. It is
+split into the two sections. The first section explains the release process,
+while the second section documents for maintainers how to make a release.
+
+- [About the Release Process](#about-the-release-process)
+    - [Historic Context](#historic-context)
+    - [Beta and Stable Releases](#beta-and-stable-releases)
+    - [Release Artifacts](#release-artifacts)
+    - [Automation](#automation)
+- [How to Cut a Release](#how-to-cut-a-release)
+    - [1. Test `rustup-init.sh`](#1-test-rustup-initsh)
+    - [2. Update Version](#2-update-version)
+    - [3. Update Checksum](#3-update-checksum)
+    - [4. Sync `master` to `stable`](#4-sync-master-to-stable)
+    - [5. Test the `beta` Release](#5-test-the-beta-release)
+    - [6. Prepare Blog Post for `stable` Release](#6-prepare-blog-post-for-stable-release)
+    - [7. Request the `stable` Release](#7-request-the-stable-release)
+    - [8. Tag the `stable` Release](#8-tag-the-stable-release)
+
+## About the Release Process
+
+This section explains the release process for Rustup.
+
+### Historic Context
+
+The Rustup release process has evolved over time. In the past, the process
+involved more manual steps and large parts of it were executed locally. This
+required careful execution and coordination to avoid accidentally overwriting
+exiting releases, since the tooling did not implement sanity checks that would
+prevent destructive actions.
+
+A full step-by-step guide for the old process can be found in
+the [old developer guide](https://github.com/rust-lang/rustup/blob/1cf1b5a6d80c978e0dcaabbce5f10b3861612425/doc/dev-guide/src/release-process.md).
+The following summarizes how release artifacts were created and copied around.
+
+In the past, both beta and stable releases were started by merging a new commit
+into the `stable` branch in [rust-lang/rustup]. This started a new build on
+GitHub Actions that produced release
+artifacts ([source](https://github.com/rust-lang/rustup/blob/1cf1b5a6d80c978e0dcaabbce5f10b3861612425/.github/workflows/ci.yaml#L144-L151)),
+which were uploaded to `s3://dev-static-rust-lang-org/rustup/dist`. As new
+commits were merged into `stable`, they would overwrite the artifacts from prior
+builds.
+
+The release artifacts were then copied to the final location by running a script
+on a local machine. This script would download the artifacts from the
+`dev-static` bucket on S3 and upload them to the `static` bucket. The script
+also generated a new manifest with the given version, and uploaded a copy of the
+files to an archive directory.
```

Comment:
This paragraph doesn't sound quite true to me - there was an intermediate step that uploaded something back to dev-static I think? Maybe just the manifest?